core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -174,7 +176,13 @@ public void commit(TableMetadata base, TableMetadata metadata) {
   @Override
   public FileIO io() {
     if (defaultFileIo == null) {
-      defaultFileIo = new HadoopFileIO(conf);
+      defaultFileIo = ClassLoaderUtil.fromProperty(
+          current() == null ? new HashMap<>() : current().properties(),
```

Comment:
We should move instantiation to a separate method that accepts `TableMetadata` so that we can use the correct metadata in transactions.